### PR TITLE
Move PaperSystem to Shared

### PR DESCRIPTION
**Pull request by Plykiya**
## About the PR

This started with me wanting to try and deal with the user seeing the localization message, then I realized I couldn't just `paper.Content = Loc.String()` during CompInit because it triggered a test error, and now I just decided to move everything to shared.

## Why / Balance

So the users don't see the localization ID when they first open the paper
I noticed this in https://github.com/space-wizards/space-station-14/pull/30298 and said I'd fix it

## Technical details

- Get rid of separated Client/Server PaperComponent and merge them on Shared so that content can be properly networked to Client
- Rename Client.PaperSystem into Client.PaperVisualizerSystem to better reflect what it actually is, update incorrect namespace
- Move Server.PaperSystem into Shared
- Update Cargo, Fax, Forensics, Labels, Nuke, Random Books, and Artifact Analyzer to use the new Shared.Paper namespace
- Dirty paper whenever `SetContent()` or `TryStamp()` is called, and on MapInit for localization

## Media

Before:

https://github.com/user-attachments/assets/0df702d9-d252-4e93-8660-cbb158669df5

After:

https://github.com/user-attachments/assets/26229af6-d833-4d27-a17b-a8f654b07b74

## Requirements

- [x] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

no one even notices this so no

### CODE REVIEW

**Comment by Plykiya** [comment]
guh, real test fail

**Comment by Plykiya** [comment]
my skill issue is greater than previously anticipated

**Comment by Plykiya** [comment]
I'd like to move the entirety of the label/hand labeler stuff over to Shared after this

File: Content.Shared/Paper/PaperSystem.cs [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+                var stampPaperOtherMessage = Loc.GetString("paper-component-action-stamp-paper-other",
+                        ("user", args.User),
+                        ("target", args.Target),
+                        ("stamp", args.Used));
+
+                _popupSystem.PopupEntity(stampPaperOtherMessage, args.User, Filter.PvsExcept(args.User, entityManager: EntityManager), true);
+                var stampPaperSelfMessage = Loc.GetString("paper-component-action-stamp-paper-self",
+                        ("target", args.Target),
+                        ("stamp", args.Used));
+                _popupSystem.PopupEntity(stampPaperSelfMessage, args.User, args.User);
```
Comment (metalgearsloth) [inline Content.Shared/Paper/PaperSystem.cs:137]: I think popupclient for this one

File: Content.Shared/Paper/PaperSystem.cs [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+                _audio.PlayPvs(stampComp.Sound, entity);
+
```
Comment (metalgearsloth) [inline Content.Shared/Paper/PaperSystem.cs:140]: playpredicted

File: Content.Shared/Paper/PaperSystem.cs [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+using Content.Shared.Popups;
+using Content.Shared.Tag;
+using Robust.Shared.Player;
+using Robust.Shared.Audio.Systems;
+using static Content.Shared.Paper.PaperComponent;
+
+namespace Content.Shared.Paper;
+
+public sealed class PaperSystem : EntitySystem
+    {
```
Comment (metalgearsloth) [inline Content.Shared/Paper/PaperSystem.cs:16]: 1 less indent plz

**Pull request merged**